README.md

```diff
@@ -1,35 +1,115 @@
+- environment variables for the compose file at `.env`
+    - some individual environment variables needed for the docker compose file
+- nginx configuration at `nginx/conf.d/default.conf`
+    - the setup of the reverse proxy which maps the requests to the correct service
+- ssl-certs at `ssl-certs/`
+    - a folder containing the ssl certifates, not present in the template
+- databases at `data/`
+    - a folder where all databases are placed, automatically created at startup
+- realm-ttemplate for keycloak at `keycloak-realm-template.json`
+    - a template realm to import for a quick start
+
+To run the project by your own beside cloning the template folder a few configuration steps need to be done. The following guides you through all steps. Also make sure you installed [docker](https://docs.docker.com/engine/install/).
+
+### 1. Place the ssl cert
+
+Create a folder called `ssl-certs` and place the ssl-certtificate and the correspondig private key there. An exemplary resulting folder structure looks like this: \
+`ssl-certs/my-domain/cert.crt` \
+`ssl-certs/my-domain/privatekey.key`
+
+### 2. Adapt the nginx configuration
+
+Open the nginx configuration file places at `nginx/conf.d/default.conf`.
+
+The head of the file looks like this:
+```
+resolver 127.0.0.11 valid=30s; # Docker nameserver
+
+server {
+    listen     80;
+
+    return 301 https://$host$request_uri;
+}
+
+server {
+    listen     443 ssl;
+
+    ssl_certificate /etc/ssl-certs/my-domain/cert.crt;
+    ssl_certificate_key /etc/ssl-certs/my-domain/privatekey.key;
+```
+
+Here you need to set the ports for http and https in the listen directive. If you user another port than 443 for https you also have to add the port to the redirect. An example with port 8000 for http and port 8443 for https looks like this:
+```
+server {
+    listen     8000;
+    return 301 https://$host:8443$request_uri;
+}
+server {
+    listen     8443 ssl;
+```
+You also need to adjust the certificate paths to match your certificates.
+If you placed your certificate files at `ssl-certs/example.com/cert.crt` at `ssl-certs/example.com/privatekey.key` the resulting lines look like this:
+```
+ssl_certificate /etc/ssl-certs/example.com/cert.crt;
+ssl_certificate_key /etc/ssl-certs/example.com/privatekey.key;
+```
+
+### 3. Adjust environment variables for the compose file
+
+In the `.env` file you need to set the url and the ports. For example if you run at `example.com` on 80 for http and 443 for https the resulting file looks like this:
+```
+url=https://example.com
+http_port=80
+https_port=443
+```
+If you run at `example.com` on 8000 for http and 8443 for https the resulting file looks like this:
+```
+url=https://example.com:8443
+http_port=8000
+https_port=8443
+```
+
+### 4. Run
+
+If you finished the above steps you can run the project with the following commands: 
 
 Run the docker containers with the following command:
 ```sh
 docker compose up -d
```

Comment:
Does that automatically connect to your real network and not only to the virtual docker network?
Also, where do you apply these changes for the docker file?

Reply:
Which docker file needs changes? It connects the reverse proxy to the host network as configured in the `docker-comnpose.yaml`